packages/babel-plugin-check-es2015-constants/src/index.js

```diff
@@ -7,7 +7,32 @@ export default function({ messages }) {
           if (binding.kind !== "const" && binding.kind !== "module") continue;
 
           for (const violation of (binding.constantViolations: Array)) {
-            throw violation.buildCodeFrameError(messages.get("readOnly", name));
+            const throwNode = t.throwStatement(
+              t.newExpression(t.identifier("Error"), [
+                t.stringLiteral(messages.get("readOnly", name)),
+              ]),
+            );
+
+            // Returns a comma expression of IIFE wrapped throwNode followed by given expression.
+            const throwBefore = expression =>
```

Comment:
I considered this and decided to keep it inside the visitor for simplicity. But I'm fine with both ways and can change it for sure.